### ISSUE TITLE
XPathMessageSplitter Not supported: http://javax.xml.XMLConstants/property/accessExternalDTD

### ISSUE THREAD

**Issue by jasonparallel** [spring-projects/spring-integration#2993]
Affects Version(s):4.3.20.RELEASE

Same issue as https://github.com/spring-projects/spring-integration/issues/2723 but using XPathMessageSplitter.

```
Caused by: java.lang.IllegalArgumentException: Not supported: http://javax.xml.XMLConstants/property/accessExternalDTD
	at org.apache.xalan.processor.TransformerFactoryImpl.setAttribute(TransformerFactoryImpl.java:571)
	at org.springframework.xml.transform.TransformerFactoryUtils.defaultSettings(TransformerFactoryUtils.java:56)
	at org.springframework.xml.transform.TransformerFactoryUtils.newInstance(TransformerFactoryUtils.java:32)
	at org.springframework.integration.xml.splitter.XPathMessageSplitter.<init>(XPathMessageSplitter.java:109)
```

**Comment by jasonparallel** [comment]
There appears to be a fix for this in 3.0.7 of spring-ws TransformerFactoryUtils but not the 2.4 line

**Comment by artembilan** [comment]
OK. Since there is not going to be a fix in Spring WS 2.4.x, I'm going to go ahead and cover all the `TransformerFactoryUtils.newInstance()` failures here in Spring Integration.
At least for the mentioned version `4.3.x`: all others are covered by Spring WS 3.0.7.

**Comment by artembilan** [comment]
@gregturn,

This is one more concern from community about unstable `TransformerFactoryUtils` logic.
May it is time to fix it in the Spring WS 2.4.x and release for upcoming Spring Boot 1.5.x ?

Meanwhile I just copy/paste your class from the `master`...

Thanks

**Comment by artembilan** [comment]
Fixed via: https://github.com/spring-projects/spring-integration/pull/2994
